middleware/grpc.go

```diff
@@ -0,0 +1,104 @@
+package middleware
+
+import (
+	"context"
+	"io"
+	"strconv"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	grpcUtils "github.com/weaveworks/common/grpc"
+	"github.com/weaveworks/common/httpgrpc"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+// PrometheusGRPCUnaryInstrumentation records duration of gRPC requests client side.
+func PrometheusGRPCUnaryInstrumentation(metric *prometheus.HistogramVec) grpc.UnaryClientInterceptor {
+	return func(ctx context.Context, method string, req, resp interface{}, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
+		start := time.Now()
+		err := invoker(ctx, method, req, resp, cc, opts...)
+		metric.WithLabelValues(method, errorCode(err)).Observe(time.Since(start).Seconds())
+		return err
+	}
+}
+
+// PrometheusGRPCStreamInstrumentation records duration of streaming gRPC requests client side.
+func PrometheusGRPCStreamInstrumentation(metric *prometheus.HistogramVec) grpc.StreamClientInterceptor {
```

Comment:
The equivalent server call in this package is `StreamServerInstrumentInterceptor`, so perhaps `StreamClientInstrumentInterceptor`?

Reply:
@bboreham Thanks, done.